libraries/standard/mqtt/include/mqtt.h

```diff
@@ -116,6 +116,17 @@ struct MQTTPubAckInfo
     MQTTPublishState_t publishState; /**< @brief The current state of the publish process. */
 };
 
+/**
+ * @brief The status codes in the SUBACK response to a subscription request.
+ */
+typedef enum MQTTSubAckStatus
+{
+    MQTTSubAckSuccessQos0 = 0x00, /**< @brief Success with a maximum delivery at QoS 0 . */
+    MQTTSubAckSuccessQos1 = 0x01, /**< @brief Success with a maximum delivery at QoS 1. */
+    MQTTSubAckSuccessQos2 = 0x02, /**< @brief Success with a maximum delivery at QoS 2. */
+    MQTTSubAckFailure = 0x80      /**< @brief Failure. */
+} MQTTSubAckStatus_t;
```

Comment:
This still seems unnecessary to me given we already have MQTTQoS_t, and list all status codes in the function documentation. Will disagree and commit since otherwise I don't have issues with this PR